test/k8s-integration/main.go

```diff
@@ -196,15 +203,31 @@ func handle() error {
 			glog.V(4).Infof("Set Kubernetes feature gates: %v", *kubeFeatureGates)
 		}
 
-		err = clusterUp(k8sDir, *gceZone)
-		if err != nil {
-			return fmt.Errorf("failed to cluster up: %v", err)
+		switch *deploymentStrat {
+		case "gce":
+			err = clusterUpGCE(k8sDir, *gceZone)
+			if err != nil {
+				return fmt.Errorf("failed to cluster up: %v", err)
+			}
+		case "gke":
+			err = clusterUpGKE(*gceZone)
+			if err != nil {
+				return fmt.Errorf("failed to cluster up: %v", err)
+			}
+		default:
+			return fmt.Errorf("deployment-strategy must be set to 'gce' or 'gke', but is: %s", *deploymentStrat)
 		}
+
 	}
 
 	if *teardownCluster {
 		defer func() {
-			err := clusterDown(k8sDir)
+			var err error
+			if *deploymentStrat == "gke" {
```

Comment:
nit: this can also be better as a switch statement and should contain a default with an error.

It's possible to have `teardownCluster` set without `bringUpCluster` set. So it's not safe to assume `*deploymentStrat != "gke" --> deploymentStrat == gce` in this case.